velox/expression/tests/CastExprTest.cpp

```diff
@@ -651,6 +658,68 @@ TEST_F(CastExprTest, invalidDate) {
       "date", {"2012-Oct-23"}, {0}, true, false, VARCHAR(), DATE());
 }
 
+TEST_F(CastExprTest, stringToDate) {
+  for (bool conf : {true, false}) {
+    setCastStringToDateNonStandard(conf);
+    testCast<std::string, int32_t>(
+        "date",
```

Comment:
wondering if we should include a case for trailing (BC) for the case where it is valid

Reply:
Added. It's invalid in presto but valid in spark.

Reply:
@bikramSingh91 My previous understanding of trailing (BC) was wrong. The trailing (BC) starts with an additional space. So `cast("2015-03-18(BC)" as date` returns NULL in spark but `cast("2015-03-18 (BC)" as date` returns `2015-03-18`